bigquery_etl/schema/__init__.py

```diff
@@ -265,7 +266,10 @@ def _traverse(
             else:
                 if update and add_missing_fields:
                     # node does not exist in schema, add to schema
-                    columns.append(node.copy())
+                    if node["type"] == "RECORD":  # deep copy record fields
+                        columns.append(ujson.loads(ujson.dumps(node)))
```

Comment:
Used `usjon` instead of `copy.deepcopy(node)` because it's faster (15ms vs 6.5ms)

Reply:
I was interested in whether I should adopt this in other projects so went to look into it, and seems like ujson is in maintenance-only mode (assuming this [link](https://github.com/ultrajson/ultrajson?tab=readme-ov-file#project-status) is the correct project) and recommending people switch to `orjson` (https://pypi.org/project/orjson/).

I guess this is probably not necessary for this change, since it seems like we already have `ujson` in the project deps, but may be worth migrating to `orjson` in the future.

Reply:
Oh I didn't know `ujson` was in maintenance-only, I just saw that it was already in the requirements.in.  `orjson` makes sense to use since the interface looks the same